Dockerfile-alpine.template

```
@@ -116,6 +116,10 @@ RUN set -eux; \
 		echo "$adapter:" >> ./config/database.yml; \
 		echo "  adapter: $adapter" >> ./config/database.yml; \
 	done; \
+{{ if ([ "5.0" ] | index(env.version)) and ([ "alpine3.21" ] | index(env.variant)) then ( -}}
+	# Fix incompatibility with sqlite 1.4 in alpine 3.21
+	sed -i 's/"sqlite3", "~> 1.4.0"/"sqlite3", "~> 1.5.0"/g' Gemfile; \
```

I try to also be explicit about `-e` on `sed` (so we're not relying on the heuristics); this is mostly harmless, but if you're already in there, might as well add this too: :eyes:

```suggestion
	sed -i -e 's/"sqlite3", "~> 1.4.0"/"sqlite3", "~> 1.5.0"/g' Gemfile; \
```

---

>if/when this no longer applies

In that regard we usually put a `grep` for the expected part so that it fails when it needs to be changed or dropped. Like [postgres](https://github.com/docker-library/postgres/blob/15778149223e016ba1bd43db00825226ddc821a4/Dockerfile-debian.template#L50-L52) and [rabbitmq](https://github.com/docker-library/rabbitmq/blob/99ed8dff13e2b79fc189b76bbbfec8d07947e6b9/Dockerfile-alpine.template#L293-L295)?

---

Yeah, that would make some sense, I think -- I bet if this line is updated, it would be to fix this same issue, but it's possible it's something less interesting like swapping the `"` to `'` in which case we'll start failing again and scratch our heads.